src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.CompilationTrackerState.cs

```diff
@@ -45,19 +44,25 @@ private abstract class CompilationTrackerState
                 /// empty, otherwise a <see cref="InProgressState"/>.
                 /// </summary>
                 public static WithCompilationTrackerState Create(
+                    bool isFrozen,
```

Comment:
Moved the frozen-bit out of the CompilationTrackerGeneratorInfo  into the tracker-state itself.  i found it much easier to reason about there.